static/js/services/enketo.js

```diff
@@ -172,27 +248,23 @@ angular.module('inboxServices').service('Enketo',
       var formContainer = wrapper.find('.container').first();
       formContainer.html(doc.html);
 
-      return EnketoPrepopulationData(doc.model, instanceData)
-        .then(function(instanceStr) {
-          var form = new EnketoForm(wrapper.find('form').first(), {
-            modelStr: doc.model,
-            instanceStr: instanceStr
-          });
-          var loadErrors = form.init();
-          if (loadErrors && loadErrors.length) {
-            return $q.reject(new Error(JSON.stringify(loadErrors)));
-          }
-          wrapper.show();
+      return getEnketoOptions(doc, instanceData).then(function(options) {
```

Comment:
Yeah. I like it for simple promises (ie: only a single link in the chain, and no `catch`), but I don't feel strongly about it if you prefer the other style.